### PR TITLE
Added Square Root Decomposition in C++ and Python3

### PR DESCRIPTION
**Pull request by kaustubh-pandey**

### CODE REVIEW

**Comment by kaustubh-pandey** [comment]
In reference to issue #562

File: Square_Root_Decomposition/Square_Root_Decomposition.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		blocks[index]+=a[i];
+	}
+	return blocks;
+}
+int main(){
+	/* Applying square root decomposition to perform range queries
+	for finding the sum of numbers in a given range 
+	Indexing : 0 based in array
+	*/
+	int n = 10; //number of elements
```
Comment (sridharjajoo) [inline Square_Root_Decomposition/Square_Root_Decomposition.cpp]: always better to calculate the size of the array by using sizeof(arr)/sizeof(arr[0])

Comment (kaustubh-pandey) [reply Square_Root_Decomposition/Square_Root_Decomposition.cpp]: I have updated the code

**Comment by kaustubh-pandey** [comment]
I have updated the code

File: Square_Root_Decomposition/Square_Root_Decomposition.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+void update(int index,int value,int *blocks,int a[],int n){
+	blocks[(int)(index/sqrt(n))] += value-a[index];
+	a[index]=value;
+}
+//assuming 0 based indexing
```
Comment (jainaman224) [inline Square_Root_Decomposition/Square_Root_Decomposition.cpp]: Add a new line above every function.
In your case add it above the comment for function.

File: Square_Root_Decomposition/Square_Root_Decomposition.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	cout<<query(0,2,blocks,a,n)<<endl; //find sum from index 0 to index 2
+	cout<<query(3,7,blocks,a,n)<<endl; //find sum from index 3 to index 7
+	cout<<query(1,4,blocks,a,n)<<endl; //find sum from index 1 to index 4
+	//update
+	update(3,1,blocks,a,n);				//update the value at index 3 with value of 1
+	//query
+	cout<<query(1,4,blocks,a,n)<<endl;	//find sum from index 1 to index 4 
+	return 0;
+}
+
```
Comment (jainaman224) [inline Square_Root_Decomposition/Square_Root_Decomposition.cpp]: Add sample input output.

**Comment by jainaman224** [comment]
@kaustubh-pandey Are you working on it?

**Comment by kaustubh-pandey** [comment]
Yeah, I will make the commit

**Comment by kaustubh-pandey** [comment]
I have made the changes

File: Square_Root_Decomposition/Square_Root_Decomposition.cpp (hunk 3 above)
Comment (jainaman224) [reply Square_Root_Decomposition/Square_Root_Decomposition.cpp]: Move to output to the last of file.

File: Square_Root_Decomposition/Square_Root_Decomposition.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
+#include<bits/stdc++.h>
```
Comment (jainaman224) [inline Square_Root_Decomposition/Square_Root_Decomposition.cpp]: Please use 4 space indentation in both files.

**Comment by kaustubh-pandey** [comment]
I have made the changes

**Pull request merged**